### PR TITLE
Downgrade cats dependency

### PR DESCRIPTION
**Pull request by ravjotbrar**
### Summary

Downgrade cats dependency

### Description

Since Spark uses an older version of cats, this change downgrades the connector's version of cats to align with Spark's. In order for our connector to be compatible with the older version of cats-core, some import statements in our utils needed to be adjusted. 

### Related Issue

https://github.com/vertica/spark-connector/issues/207

### Additional Reviewers

@alexey-temnikov 
@alexr-bq 
@jonathanl-bq

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -78,6 +78,20 @@ Once in the examples directory, run:
...
+### Using Thin Jar
+If you are using the thin jar and running into an error similar to the following:
+`java.lang.NoSuchMethodError: 'void cats.kernel.CommutativeSemigroup.$init$(cats.kernel.CommutativeSemigroup)'`, you may need to shade the cats dependency in your project.
+
+This can be done by adding the following to your build.sbt file:
+`assembly / assemblyShadeRules := {
+val shadePackage = "com.azavea.shaded.demo"
+Seq(
+ShadeRule.rename("cats.kernel.**" -> s"$shadePackage.cats.kernel.@1").inAll
+)
```
Comment (jonathanl-bq) [inline examples/README.md:90]: nit: indentation

**Review by jonathanl-bq — APPROVED** [review]

File: examples/README.md (hunk 1 above)
Comment (ravjotbrar) [reply examples/README.md]: Fixed

File: connector/src/main/scala/com/vertica/spark/util/cleanup/CleanupUtils.scala [hunk 2]
```diff
@@ -13,7 +13,7 @@
 
 package com.vertica.spark.util.cleanup
 
-import cats.implicits.toTraverseOps
+import cats.implicits._
```
Comment (alexey-temnikov) [inline connector/src/main/scala/com/vertica/spark/util/cleanup/CleanupUtils.scala:16]: Why importing everything? Best practise would be to keep more specific imports.

**Review by alexr-bq — APPROVED** [review]

File: connector/src/main/scala/com/vertica/spark/util/cleanup/CleanupUtils.scala (hunk 2 above)
Comment (ravjotbrar) [reply connector/src/main/scala/com/vertica/spark/util/cleanup/CleanupUtils.scala:16]: Based on my research, importing cats.implicits._ is recommended.

Comment (ravjotbrar) [reply connector/src/main/scala/com/vertica/spark/util/cleanup/CleanupUtils.scala:16]: https://svejcar.dev/posts/2020/04/14/scala-cats-howto-guide/

**Pull request merged**